libraries/AP_RPM/RPM_ESC_Telem.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_ESC_Telem/AP_ESC_Telem.h>
+#include <AP_Vehicle/AP_Vehicle.h>
+#include <SRV_Channel/SRV_Channel.h>
+#include <AP_Motors/AP_Motors.h>
+#if APM_BUILD_TYPE(APM_BUILD_ArduPlane)
+#include "../ArduPlane/quadplane.h"
+#endif
+
+#include "RPM_ESC_Telem.h"
+
+extern const AP_HAL::HAL& hal;
+
+//#define DEBUG_RPM_ESC
+
+/*
+   open the sensor in constructor
+*/
+AP_RPM_ESC_Telem::AP_RPM_ESC_Telem(AP_RPM &_ap_rpm, uint8_t _instance, AP_RPM::RPM_State &_state,uint8_t _rpm_driver_type) :
+    AP_RPM_Backend(_ap_rpm, _instance, _state)
+{
+    instance = _instance;
+    rpm_driver_type = _rpm_driver_type;
+}
+
+// convert a param motor mask to an uint32_t channel mask
+uint32_t AP_RPM_ESC_Telem::param_motor_mask_to_channel_mask(const char* param_name) const
+{
+    float motor_mask;
+    if (AP_Param::get(param_name,motor_mask)) {
```

Comment:
much too slow to be doing on main thread at runtime
could use a param pointer, or get once at startup